source/base/ByteBuffer.ooc

```diff
@@ -50,11 +54,14 @@ ByteBuffer: class {
 	copyTo: func (other: This, start: Int, destination: Int, length: Int) {
 		memcpy(other pointer + destination, this pointer + start, length)
 	}
-	new: static func ~size (size: Int) -> This { _RecyclableByteBuffer new(size) }
+	new: static func ~size (size: Int, allocator := Allocator defaultAllocator()) -> This { _RecyclableByteBuffer new(size, allocator) }
 	new: static func ~recover (pointer: Byte*, size: Int, recover: Func (This) -> Bool) -> This {
 		_RecoverableByteBuffer new(pointer, size, recover)
 	}
-	free: static func ~all { _RecyclableByteBuffer _free~all() }
+	free: static func ~all {
+		_RecyclableByteBuffer _free~all()
+		Allocator free~all()
```

Comment:
Since it would be possible to use `Allocator` for other purposes besides `ByteBuffer`, I'd suggest not freeing it here but rather adding its own explicit register call to `GlobalCleanup`.
...the only bothersome thing is that the order is important (it needs to happen after `ByteBuffer free~all()`). 
`GlobalCleanup register(|| Allocator free~all(), true)` puts the call at the bottom of the (current) stack. If all else fails just put it after line 67 of this file.